### PR TITLE
multisites and router swapping

### PR DESCRIPTION
**Pull request by qadan**

### CODE REVIEW

File: gearman-worker.sh [hunk 1]
```diff
@@ -4,12 +4,19 @@
...
+  for site in $SITE_URI_LIST; do
+    GEARMAN_FUNCTIONS="${GEARMAN_FUNCTIONS} `$DRUSH --root=$DRUPAL_ROOT --uri=$site islandora-job-list-functions --param_format`"
+    DRUSH_RETURN=$?
+    if [ $DRUSH_RETURN -ne 0 ]; then
+      logger "Could not determine worker functions."
+      exit 1
+    fi
+  done
+else
+  GEARMAN_FUNCTIONS=${GEARMAN_FUNCTIONS[@]/#/-f }
```
Comment (adam-vessey) [inline gearman-worker.sh]: Why go to a string?

File: gearman-worker.sh [hunk 2]
```diff
@@ -4,12 +4,19 @@
 
 # Get a Bash array containing a list of all the job names.
 if [ -z ${GEARMAN_FUNCTIONS:+set} ]; then
-  GEARMAN_FUNCTIONS=(`$DRUSH --root=$DRUPAL_ROOT php-eval --format=string "return implode('$IFS', array_keys(module_invoke_all('islandora_job_register_jobs')));"`)
-  DRUSH_RETURN=$?
-  if [ $DRUSH_RETURN -ne 0 ]; then
-    logger "Could not determine worker functions."
-    exit 1
-  fi
+  GEARMAN_FUNCTIONS=""
```
Comment (adam-vessey) [inline gearman-worker.sh]: Could initialize as an empty array:
```bash
declare -a GEARMAN_FUNCTIONS
```

File: gearman-worker.sh [hunk 3]
```diff
@@ -4,12 +4,19 @@
...
+  # XXX: STDOUT from Drush commands using @sites comes with an unfortunate
+  # amount of unfriendly baggage; better to loop manually.
+  for site in $SITE_URI_LIST; do
+    GEARMAN_FUNCTIONS="${GEARMAN_FUNCTIONS} `$DRUSH --root=$DRUPAL_ROOT --uri=$site islandora-job-list-functions --param_format`"
```
Comment (adam-vessey) [inline gearman-worker.sh]: And then append to the array:
```bash
GEARMAN_FUNCTIONS+=(`$DRUSH --root=$DRUPAL_ROOT --uri=$site islandora-job-list-functions`)
```
... I think? Could then allow the replacement below, with the `-f ` prefixing?

File: gearman-worker.sh [hunk 4]
```diff
@@ -4,12 +4,19 @@
...
+    DRUSH_RETURN=$?
+    if [ $DRUSH_RETURN -ne 0 ]; then
+      logger "Could not determine worker functions."
+      exit 1
+    fi
+  done
 fi
 
-exec $GEARMAN_BIN -v -h $GEARMAN_HOST -p $GEARMAN_PORT -w ${GEARMAN_FUNCTIONS[@]/#/-f } -- $DRUSH --root=$DRUPAL_ROOT -u 1 islandora-job-router
+GEARMAN_FUNCTIONS=${GEARMAN_FUNCTIONS[@]/#/-f }
```
Comment (adam-vessey) [inline gearman-worker.sh]: Reusing variables like this is... weird.

File: README.md [hunk 5]
```diff
@@ -36,7 +38,9 @@ Copy the `upstart/gearman-workers.conf` and `upstart/gearman-worker.conf` files
 The workers should start and stop automatically at boot/shutdown (currently following/preceding `mysql` of startup/shutdown; may be desirable to make into configuration at some point?).
 
 To start the workers immediately:
-```# start gearman-workers```
+```bash
+start gearman-workers
```
Comment (adam-vessey) [inline README.md]: Just a note: The `#` bits were intentional here, to indicate a root/super-user prompt... I'm fine getting rid of 'em, just sayin'.

Comment (qadan) [reply README.md]: yeaaaah just try'na make it consistent with formatting between prompt/script by using the `bash` notation, but then `#` makes it into a comment :/

File: gearman-defaults.sh [hunk 6]
```diff
@@ -14,3 +14,7 @@ GEARMAN_HOST=${GEARMAN_HOST:-`$DRUSH --root=$DRUPAL_ROOT variable-get islandora_
 GEARMAN_PORT=${GEARMAN_PORT:-`$DRUSH --root=$DRUPAL_ROOT variable-get islandora_job_server_port --exact --format=string 2> /dev/null || echo -n '4730'`}
 GEARMAN_BIN=${GEARMAN_BIN:-`which gearman`}
 GEARMAN_USER=${GEARMAN_USER:-www-data}
+# For multisites requiring per-site routing, a different router is provided
+# at https://github.com/discoverygarden/gearman-multisite-job-router
+ROUTER=${ROUTER:-"$DRUSH --root=$DRUPAL_ROOT -u 1 islandora-job-router"}
+SITE_URI_LIST=${SITE_URI_LIST:-`$drush --root=$DRUPAL_ROOT @sites status --interactive --field-labels=0 --fields=uri | sed -z 's/\n//g'`}
```
Comment (adam-vessey) [inline gearman-defaults.sh]: Is goof: `$drush` vs `$DRUSH`...

**Pull request merged**